### PR TITLE
Implement converter for esriTS symbols for featureServers

### PR DESCRIPTION
**Pull request by merydian**
## Description

The file qgsarcgisrestutils.cpp curently does not support conversions for esriTS type symbols. This pull request adds functionality to convert esriTS symbols in ArcGis featureServers to a QgsMarkerSymbol. When loading featureServers with this symbol type, symbols will now load automatically.

### CODE REVIEW

File: src/core/providers/arcgis/qgsarcgisrestutils.cpp [hunk 1]
```diff
@@ -748,6 +747,84 @@ std::unique_ptr<QgsMarkerSymbol> QgsArcGisRestUtils::parseEsriPictureMarkerSymbo
...
+  QString chr;
+  chr = symbolData.value( QStringLiteral( "text" ) ).toString();
+
+  double pointSize;
+  pointSize = symbolData.value( QStringLiteral( "font" ) ).toMap().value( QStringLiteral( "size" ) ).toDouble();
+
+  QColor color = convertColor( symbolData.value( QStringLiteral( "color" ) ) );
+
+  double angle;
+  angle = symbolData.value( QStringLiteral( "angle" ) ).toDouble();
```
Comment (nyalldawson) [inline src/core/providers/arcgis/qgsarcgisrestutils.cpp:766]:
```suggestion
  const QString fontFamily = symbolData.value( QStringLiteral( "font" ) ).toMap().value( QStringLiteral( "family" ) ).toString();

  const QString chr = symbolData.value( QStringLiteral( "text" ) ).toString();

  const double pointSize = symbolData.value( QStringLiteral( "font" ) ).toMap().value( QStringLiteral( "size" ) ).toDouble();

  const QColor color = convertColor( symbolData.value( QStringLiteral( "color" ) ) );

  const double angle = symbolData.value( QStringLiteral( "angle" ) ).toDouble();
```

Comment (nyalldawson) [inline src/core/providers/arcgis/qgsarcgisrestutils.cpp:766]: It's tricky to confirm without a server to check against, but I suspect we need to transform the rotation here. From the ESRI docs: "The rotation is from East in a counter-clockwise direction where East is the 0° axis.", whereas QGIS symbol markers use clockwise rotation from north. So we'll need to do a "90-angle from JSON" modification here.

File: src/core/providers/arcgis/qgsarcgisrestutils.cpp [hunk 2]
```diff
@@ -748,6 +747,84 @@ std::unique_ptr<QgsMarkerSymbol> QgsArcGisRestUtils::parseEsriPictureMarkerSymbo
...
+  const QString fontStyle = symbolData.value( QStringLiteral( "font" ) ).toMap().value( QStringLiteral( "style" ) ).toString();
+  markerLayer->setFontStyle( fontStyle );
+
+  double xOffset = symbolData.value( QStringLiteral( "xoffset" ) ).toDouble();
+  double yOffset = symbolData.value( QStringLiteral( "yoffset" ) ).toDouble();
+
+  markerLayer->setOffset( QPointF( xOffset, yOffset ) );
+  markerLayer->setOffsetUnit( Qgis::RenderUnit::Points );
+
+  markerLayer->setSizeUnit( Qgis::RenderUnit::Points );
```
Comment (nyalldawson) [inline src/core/providers/arcgis/qgsarcgisrestutils.cpp:785]: We also need to set the stroke width unit to points too

File: tests/src/python/test_provider_ogr.py [hunk 3]
```diff
@@ -3551,7 +3551,7 @@ def testExtentCsv(self):
         with open(datasource_2d, 'w') as f:
             f.write('id,WKT\n')
             for i in range(9):
-                f.write(f'{i},POINT ({2*i} {i-3})\n')
+                f.write(f'{i},POINT ({2 * i} {i - 3})\n')
```
Comment (nyalldawson) [inline tests/src/python/test_provider_ogr.py:3554]: Can you revert the changes to these other files please?

File: tests/src/core/testqgsarcgisrestutils.cpp [hunk 4]
```diff
@@ -253,6 +253,69 @@ void TestQgsArcGisRestUtils::testParseMarkerSymbol()
...
+                              "\"haloColor\": ["
+                              "0,"
+                              "255,"
+                              "0,"
+                              "255"
+                              "],"
+                              "\"verticalAlignment\": \"bottom\","
+                              "\"horizontalAlignment\": \"left\","
+                              "\"rightToLeft\": false,"
+                              "\"angle\": 0,"
```
Comment (nyalldawson) [inline tests/src/core/testqgsarcgisrestutils.cpp:289]:
```suggestion
                              "\"angle\": 45,"
```

That'll allow us to check the angle conversion logic too

File: tests/src/core/testqgsarcgisrestutils.cpp [hunk 5]
```diff
@@ -253,6 +253,69 @@ void TestQgsArcGisRestUtils::testParseMarkerSymbol()
...
+  QgsFontMarkerSymbolLayer *fontMarkerLayer = dynamic_cast< QgsFontMarkerSymbolLayer * >( fontMarker->symbolLayer( 0 ) );
+  QVERIFY( fontMarkerLayer );
+  QCOMPARE( fontMarkerLayer->fontStyle(), QString( "normal" ) );
+  QCOMPARE( fontMarkerLayer->fontFamily(), QString( "Arial" ) );
+  QCOMPARE( fontMarkerLayer->offset(), QPointF( 0, 0 ) );
+  QCOMPARE( fontMarkerLayer->angle(), 0 );
+  QCOMPARE( fontMarkerLayer->horizontalAnchorPoint(),  QgsMarkerSymbolLayer::HorizontalAnchorPoint::Left );
+  QCOMPARE( fontMarkerLayer->verticalAnchorPoint(),  QgsMarkerSymbolLayer::VerticalAnchorPoint::Bottom );
+  QColor strokeColor = fontMarkerLayer->strokeColor();
+  QCOMPARE( strokeColor.red(), 255 );
```
Comment (nyalldawson) [inline tests/src/core/testqgsarcgisrestutils.cpp:315]: I'd go instead:
```suggestion
  QColor strokeColor = fontMarkerLayer->strokeColor();
  QCOMPARE( strokeColor.name(), QStringLiteral("#ff0000" ) );
```

We also need a check for the main color too, in addition to the stroke check

**Comment by nyalldawson** [comment]
Nice work @merydian -- this is nearly ready!

File: tests/src/core/testqgsarcgisrestutils.cpp (hunk 5 above)
Comment (merydian) [reply tests/src/core/testqgsarcgisrestutils.cpp:315]: Done

**Review by nyalldawson — APPROVED** [review]

File: src/core/providers/arcgis/qgsarcgisrestutils.cpp [hunk 6]
```diff
@@ -748,6 +747,83 @@ std::unique_ptr<QgsMarkerSymbol> QgsArcGisRestUtils::parseEsriPictureMarkerSymbo
...
+  markerLayer->setStrokeWidthUnit( Qgis::RenderUnit::Points );
+
+  QgsMarkerSymbolLayer::HorizontalAnchorPoint hAlign;
```
Comment (nyalldawson) [inline src/core/providers/arcgis/qgsarcgisrestutils.cpp:786]:
```suggestion
  QgsMarkerSymbolLayer::HorizontalAnchorPoint hAlign = QgsMarkerSymbolLayer::HorizontalAnchorPoint::HCenter;
```

File: src/core/providers/arcgis/qgsarcgisrestutils.cpp [hunk 7]
```diff
@@ -748,6 +747,83 @@ std::unique_ptr<QgsMarkerSymbol> QgsArcGisRestUtils::parseEsriPictureMarkerSymbo
...
+  QgsMarkerSymbolLayer::VerticalAnchorPoint vAlign;
```
Comment (nyalldawson) [inline src/core/providers/arcgis/qgsarcgisrestutils.cpp:787]:
```suggestion
  QgsMarkerSymbolLayer::VerticalAnchorPoint vAlign = QgsMarkerSymbolLayer::VerticalAnchorPoint::VCenter;
```

**Comment by nyalldawson** [comment]
Nice work @merydian ! Welcome to core QGIS development! :partying_face:
